src/error.js

```diff
@@ -319,14 +319,19 @@ function reportErrorToServer(message, filename, line, col, error) {
   const data = getErrorReportData(message, filename, line, col, error,
       hasNonAmpJs);
   if (data) {
-    // Report the error to viewer if it has the capability. The data passed
-    // to the viewer is exactly the same as the data passed to the server
-    // below.
-    maybeReportErrorToViewer(this, data);
     reportingBackoff(() => {
-      const xhr = new XMLHttpRequest();
-      xhr.open('POST', urls.errorReporting, true);
-      xhr.send(JSON.stringify(data));
+      // Report the error to viewer if it has the capability. The data passed
+      // to the viewer is exactly the same as the data passed to the server
+      // below.
+      maybeReportErrorToViewer(this, data).then(reportedErrorToViewer => {
+        if (!reportedErrorToViewer) {
+          // Only report to StackDriver if it's not reported to viewer, as
+          // such XHRs will trigger CSP violation for AMP4EMAIL.
```

Comment:
I don't think we should mention `AMP4EMAIL` here, since the error reporting viewer capability is designed to be a generic thing that can be used for different purposes.

Reply:
Just removed the comment. I feel that's already indicated by the name `maybeReportErrorToViewer` and have an if-else:
if not reported to viewer, reported to somewhere else (stackdriver).